torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1126,7 +1126,12 @@ ProcessGroupNCCL::~ProcessGroupNCCL() {
 
   // Abort communicators after all threads have exited to avoid having the
   // threads dying due to aborted communicator and raising a SIGABRT
-  std::string abortReason = c10::str("Process Group destroyed on rank ", rank_);
+  // We need to include PG information in the abort reason so we can tell the
+  // abort order.
+  std::string abortReason = c10::str("Process Group destroyed: ", logPrefix());
+  LOG(INFO)
```

Comment:
hmm, on second thought, this set of logs will be seen by everybody on normal exit.  its really helpful when theres a hang but it might annoy people too

Reply:
that is true.. but dtor is called only once so it should not spam the logs?